### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by bdnnk**

### CODE REVIEW

File: app/test_main.py [hunk 1]
```diff
@@ -1,3 +1,121 @@
...
+    [
+        (28, 28, [3, 2])
+    ]
+)
+def test_slightly_above_the_second_threshold(
+        cat_age: int,
+        dog_age: int,
+        expected: list[int]
+) -> None:
+    assert get_human_age(cat_age, dog_age) == expected
```
Comment (Dimosphen1) [inline app/test_main.py:121]: Remove duplicated tests. Add tests that check against unexpected arguments when you pass dict, string, tuple, list, etc. instead of the int. You could allocate separate function for that

File: app/test_main.py [hunk 2]
```diff
@@ -1,3 +1,121 @@
 from app.main import get_human_age
+import pytest
 
-# write your code here
+
+@pytest.mark.parametrize(
+    "cat_age, dog_age, expected",
+    [
+        (0, 0, [0, 0]),
```
Comment (Dimosphen1) [inline app/test_main.py:8]: Add test case that checks negative numbers as well

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested

File: app/main.py
Comment (Dimosphen1) [inline app/main.py:1]: Do not edit this file, make sure that it is absent in the PR (pull request)

File: app/test_main.py [hunk 3]
```diff
@@ -1,3 +1,134 @@
...
+        ({2}, "string"),
+        ({3: 1}, (2))
+    ]
+)
+def test_non_integer_values(
+        cat_age: int,
+        dog_age: int
+) -> None:
+    with pytest.raises(TypeError):
+        get_human_age(cat_age, dog_age)
```
Comment (Dimosphen1) [inline app/test_main.py:134]: Leave one function that is dedicated to testing errors, do not adjust the `get_human_age` code

File: app/test_main.py [hunk 4]
```diff
@@ -1,3 +1,134 @@
 from app.main import get_human_age
+import pytest
```
Comment (Dimosphen1) [inline app/test_main.py:2]: Order and group your imports to follow PEP-8

File: app/test_main.py [hunk 5]
```diff
@@ -1,3 +1,134 @@
...
+    [
+        (28, 28, [3, 2])
+    ]
+)
+def test_slightly_above_the_second_threshold(
+        cat_age: int,
+        dog_age: int,
+        expected: list[int]
+) -> None:
+    assert get_human_age(cat_age, dog_age) == expected
```
Comment (Dimosphen1) [inline app/test_main.py:100]: Remove these functions and leave only one function with multiple test cases that are passed to the `pytest.mark.parametrize` decorator

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested

File: app/test_main.py [hunk 6]
```diff
@@ -1,3 +1,40 @@
...
+
+
+@pytest.mark.parametrize(
+    "cat_age, dog_age",
+    [
+        ("cat", "dog"),
+        ("11", "12")
```
Comment (Dimosphen1) [inline app/test_main.py:32]: Add data types in addition to strings to test against, e.g. lists, dicts, tuples, etc.

File: app/test_main.py [hunk 7]
```diff
@@ -1,3 +1,40 @@
...
+    [
+        (0, 0, [0, 0]),
+        (14, 14, [0, 0]),
+        (15, 15, [1, 1]),
+        (16, 16, [1, 1]),
+        (23, 23, [1, 1]),
+        (24, 24, [2, 2]),
+        (28, 28, [3, 2]),
+        (-1, -1, [0, 0]),
+        (-35, -45, [0, 0])
```
Comment (Dimosphen1) [inline app/test_main.py:17]: Add test cases to check against numbers that have different input values for `cat_age` and `dog_age` to make sure that values do not interfere with one another

File: app/test_main.py [hunk 8]
```diff
@@ -1,3 +1,40 @@
...
+@pytest.mark.parametrize(
+    "cat_age, dog_age",
+    [
+        ("cat", "dog"),
+        ("11", "12")
+    ]
+)
+def test_invalid_values(
+        cat_age: int,
+        dog_age: int
```
Comment (Dimosphen1) [inline app/test_main.py:37]: Change the type annotation to indicate that `Any` data type could be passed here

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested
